exp/controllers/azuremanagedmachinepool_controller.go

```diff
@@ -168,6 +168,13 @@ func (r *AzureManagedMachinePoolReconciler) Reconcile(req ctrl.Request) (_ ctrl.
 		return reconcile.Result{}, errors.Errorf("failed to create scope: %+v", err)
 	}
 
+	// Always patch when exiting so we can persist changes to finalizers and status
+	defer func() {
```

Comment:
We were not making sure to patch the `AzureManagedMachinePool` at the end of reconciliation, thus they would never get to ready.

Reply:
nice catch, I think I hit this testing #822 but didn't figure out why they weren't becoming ready.